helper/restore_helper.go

```diff
@@ -47,8 +48,35 @@ var (
 type RestoreReader struct {
 	bufReader  *bufio.Reader
 	seekReader io.ReadSeeker
+	pluginCmd  *pluginCmd
 	readerType ReaderType
-	errBuf     bytes.Buffer
+}
+
+// Log plugin's stderr as warnings.
+func (r *RestoreReader) logPluginStderr() {
```

Comment:
do we actually need `logPluginStderr` and `endPluginProcess` as separate functions? they are called only from `finalizeReaderPlugin`. Can't we inline them into `finalizeReaderPlugin` in order not to spawn excessive entities?